ci/loadWin.groovy

```diff
@@ -1,3 +1,60 @@
+// Check if we can delete c:\PR-XXXX only if jenkins workspace does not exists for the PR, thus commit was merged or removed.
+def cleanup_directories() {
+    println "Cleaning pr-xxxx directories from node: NODE_NAME = ${env.NODE_NAME}"
+    def command = 'ls c:\\Jenkins\\workspace | grep -oE ".*(PR-[0-9]*)$" | sed -n -E "s/(ovms_oncommit_|ovms_ovms-windows_)//p'
+    def status = bat(returnStatus: true, script: command)
+    if ( status != 0) {
+        error "Error: trying to list jenkins workspaces."
+    }
+    def existing_wr_string = bat(returnStatus: false, returnStdout: true, script: command)
+
+    println existing_wr_string
+    def existing_wr = existing_wr_string.split(/\n/)
+
+    command = 'ls c:\\ | grep -oE "(pr-[0-9]*)$"'
+    status = bat(returnStatus: true, script: command)
+    if ( status != 0) {
+        println "No PR-XXXX detected for cleanup."
+        return
+    }
+
+    def existing_pr_string = bat(returnStatus: false, returnStdout: true, script: command)
+
+    println existing_pr_string
+    def existing_pr = existing_pr_string.split(/\n/)
+    
+    // Compare workspace with c:\pr-xxxx
+    for (int i = 0; i < existing_pr.size(); i++) {
+        def found = false
+        for (int j = 0; j < existing_wr.size(); j++) {
+            if (existing_pr[i].toLowerCase() == existing_wr[j].toLowerCase()) {
+                found = true
+                break
+            }
+            // Part of output contains the command that was run
+            if (existing_pr[i].toLowerCase().contains("grep")) {
+                found = true
+                break
```

Comment:
Looks hacky 😕

Reply:
Unfortunately no other way to do it works.